File: Hash Table/205. Isomorphic Strings/isomorphic.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        mapS = {}
        mapT = {}

        for i in range(0, len(s)):
            charS = s[i]
            charT = t[i]

            if charS in mapS.keys():
                if mapS[charS] != charT:
                    return False

            else:
                mapS[charS] = charT

            if charT in mapT.keys():
                if mapT[charT] != charS:
                    return False
            else:
                mapT[charT] = charS

        return True

File: 4. Hash Table/205. Isomorphic Strings/isomorphic.cpp
class Solution
{
public:
    bool isIsomorphic(string s, string t)
    {
        unordered_map<char, char> mapS;
        unordered_map<char, char> mapT;

        for (int i = 0; i < s.size(); i++)
        {
            char charS = s[i];
            char charT = t[i];

            if (mapS.find(charS) != mapS.end())
            {
                if (mapS[charS] != charT)
                {
                    return false;
                }
            }
            else
            {
                mapS[charS] = charT;
            }

            if (mapT.find(charT) != mapT.end())
            {
                if (mapT[charT] != charS)
                {
                    return false;
                }
            }
            else
            {
                mapT[charT] = charS;
            }
        }
        return true;
    }
};